### PR TITLE
Fix toc generation for api docs

### PR DESCRIPTION
**Pull request by vishesh92**
### Description

This PR modifies the categories of the commands API Documentation.

<details><summary>Details</summary>
<p>

|Command|Category before changes|Category after changes|
|---|---|---|
|addAccountToProject|Account|Project|
|addBaremetalHost|Host|Baremetal|
|addGloboDnsHost|Host|Network|
|addImageStoreS3|S3|Image Store|
|addNetscalerLoadBalancer|Load Balancer|Netscaler|
|addNetworkDevice|Network|Network Device|
|addObjectStoragePool|Storage Pool|Object Store|
|addPaloAltoFirewall|Firewall|Palo Alto|
|addSecondaryStorage|Host|Image Store|
|addStratosphereSsp| Stratosphere SSP|Misc Network Service Providers|
|addSwift|Swift|Image Store|
|addTrafficMonitor|Usage|Network|
|addTrafficType|Usage|Network|
|addTungstenFabricNetworkGatewayToLogicalRouter|Router|Tungsten|
|addUserToProject|User|Project|
|addVirtualMachinesToKubernetesCluster|Virtual Machine|Kubernetes Service|
|addVpnUser|User|VPN|
|cancelShutdown|Shutdown|Management|
|changeSharedFileSystemDiskOffering|Disk Offering|Shared FileSystem|
|changeSharedFileSystemServiceOffering|Service Offering|Shared FileSystem|
|configureInternalLoadBalancerElement|Load Balancer|Internal LB|
|configureNetscalerLoadBalancer|Load Balancer|Netscaler|
|configurePaloAltoFirewall|Firewall|Palo Alto|
|createInternalLoadBalancerElement|Load Balancer|Internal LB|
|createKubernetesCluster|Cluster|Kubernetes Service|
|createManagementNetworkIpRange|Network|Pod|
|createProjectRole|Role|Project|
|createProjectRolePermission|Role|Project|
|createSecondaryStorageSelector|Host|Image Store|
|createSnapshotFromVMSnapshot|Virtual Machine|Snapshot|
|createTungstenFabricFirewallPolicy|Firewall|Tungsten|
|createTungstenFabricFirewallRule|Firewall|Tungsten|
|createTungstenFabricLogicalRouter|Router|Tungsten|
|createTungstenFabricManagementNetwork|Network|Tungsten|
|createTungstenFabricPublicNetwork|Network|Tungsten|
|createVMSnapshot|Virtual Machine|Snapshot|
|createVPCOffering|VPC|VPC Offering|
|dedicateHost|Host|Dedicate Resources|
|deleteAccountFromProject|Account|Project|
|deleteKubernetesCluster|Cluster|Kubernetes Service|
|deleteManagementNetworkIpRange|Network|Pod|
|deleteNetscalerControlCenter|Load Balancer|Netscaler|
|deleteNetscalerLoadBalancer|Load Balancer|Netscaler|
|deleteNetworkDevice|Network|Network Device|
|deleteObjectStoragePool|Storage Pool|Object Store|
|deletePaloAltoFirewall|Firewall|Palo Alto|
|deleteProjectRole|Role|Project|
|deleteProjectRolePermission|Role|Project|
|deleteSslCert|Load Balancer|SSL Certificates|
|deleteStratosphereSsp| Stratosphere SSP|Misc Network Service Providers|
|deleteTrafficMonitor|Usage|Network|
|deleteTrafficType|Usage|Network|
|deleteTungstenFabricFirewallPolicy|Firewall|Tungsten|
|deleteTungstenFabricFirewallRule|Firewall|Tungsten|
|deleteTungstenFabricLogicalRouter|Router|Tungsten|
|deleteUserData|User|User Data|
|deleteUserFromProject|User|Project|
|deleteVMSnapshot|Virtual Machine|Snapshot|
|deleteVPCOffering|VPC|VPC Offering|
|disableOutOfBandManagementForHost|Host|Out-of-band Management|
|enableOutOfBandManagementForHost|Host|Out-of-band Management|
|getApiLimit|Limit|Configuration|
|getKubernetesClusterConfig|Cluster|Kubernetes Service|
|getSolidFireAccountId|Account|SolidFire|
|getSolidFireVolumeAccessGroupIds|Volume|SolidFire|
|getSolidFireVolumeSize|Volume|SolidFire|
|getVolumeSnapshotDetails|Volume|Snapshot|
|linkUserDataToTemplate|Template|User Data|
|listBrocadeVcsDeviceNetworks|Network|Brocade VCS|
|listHostsMetrics|Host|Metrics|
|listHypervisors|Hypervisor|Configuration|
|listInternalLoadBalancerElements|Load Balancer|Internal LB|
|listInternalLoadBalancerVMs|Virtual Machine|Internal LB|
|listKubernetesClusters|Cluster|Kubernetes Service|
|listManagementServersMetrics|Metrics|Management|
|listNetscalerControlCenter|Load Balancer|Netscaler|
|listNetscalerLoadBalancerNetworks|Load Balancer|Netscaler|
|listNetscalerLoadBalancers|Load Balancer|Netscaler|
|listNetworkDevice|Network|Network Device|
|listNetworkIsolationMethods|ISO|Network|
|listNiciraNvpDeviceNetworks|Network|Nicira NVP|
|listObjectStoragePools|Storage Pool|Object Store|
|listPaloAltoFirewallNetworks|Firewall|Palo Alto|
|listPaloAltoFirewalls|Firewall|Palo Alto|
|listProjectAccounts|Account|Project|
|listProjectRolePermissions|Role|Project|
|listProjectRoles|Role|Project|
|listResourceLimits|Limit|Resource Limit|
|listSecondaryStorageSelectors|Host|Image Store|
|listSslCerts|Load Balancer|SSL Certificates|
|listSwifts|Swift|Image Store|
|listTemplateDirectDownloadCertificates|Template|Certificate|
|listTrafficMonitors|Usage|Network|
|listTrafficTypeImplementors|Usage|Network|
|listTrafficTypes|Usage|Network|
|listTungstenFabricFirewallPolicy|Firewall|Tungsten|
|listTungstenFabricFirewallRule|Firewall|Tungsten|
|listTungstenFabricLogicalRouter|Router|Tungsten|
|listTungstenFabricNetwork|Network|Tungsten|
|listUsageServerMetrics|Usage|Metrics|
|listUserData|User|User Data|
|listVMSnapshot|Virtual Machine|Snapshot|
|listVPCOfferings|VPC|VPC Offering|
|listVolumesMetrics|Volume|Metrics|
|listVpnUsers|User|VPN|
|listZonesMetrics|Zone|Metrics|
|migrateResourceToAnotherSecondaryStorage|Host|Image Store|
|migrateSecondaryStorageData|Host|Image Store|
|moveNetworkAclItem|Network|Network ACL|
|prepareForShutdown|Shutdown|Management|
|provisionTemplateDirectDownloadCertificate|Template|Certificate|
|readyForShutdown|Shutdown|Management|
|registerNetscalerControlCenter|Registration|Netscaler|
|registerNetscalerServicePackage|Registration|Netscaler|
|registerOauthProvider|Registration|Oauth|
|registerUserData|User|User Data|
|releaseDedicatedHost|Host|Dedicate Resources|
|removeSecondaryStorageSelector|Host|Image Store|
|removeTungstenFabricNetworkGatewayFromLogicalRouter|Router|Tungsten|
|removeVirtualMachinesFromKubernetesCluster|Virtual Machine|Kubernetes Service|
|removeVpnUser|User|VPN|
|resetApiLimit|Limit|Configuration|
|restoreVolumeFromBackupAndAttachToVM|Virtual Machine|Volume|
|revokeTemplateDirectDownloadCertificate|Template|Certificate|
|scaleKubernetesCluster|Cluster|Kubernetes Service|
|startInternalLoadBalancerVM|Virtual Machine|Internal LB|
|startKubernetesCluster|Cluster|Kubernetes Service|
|stopInternalLoadBalancerVM|Virtual Machine|Internal LB|
|stopKubernetesCluster|Cluster|Kubernetes Service|
|triggerShutdown|Shutdown|Management|
|updateObjectStoragePool|Storage Pool|Object Store|
|updateProjectRole|Role|Project|
|updateProjectRolePermission|Role|Project|
|updateResourceLimit|Limit|Resource Limit|
|updateSecondaryStorageSelector|Host|Image Store|
|updateStorageCapabilities|Configuration|Storage Pool|
|updateTrafficType|Usage|Network|
|updateVMAffinityGroup|Virtual Machine|Affinity Group|
|updateVPCOffering|VPC|VPC Offering|
|upgradeKubernetesCluster|Cluster|Kubernetes Service|
|uploadSslCert|Load Balancer|SSL Certificates|
|uploadTemplateDirectDownloadCertificate|Template|Certificate|

</p>
</details> 

### Types of changes

- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] Enhancement (improves an existing feature and functionality)
- [ ] Cleanup (Code refactoring and cleanup, that may add test cases)
- [ ] build/CI
- [ ] test (unit or integration test code)

#### Feature/Enhancement Scale

- [ ] Major
- [x] Minor

### CODE REVIEW

**Review by sureshanaparti — APPROVED** [review]
clgtm

@DaanHoogland @rajujith  check/confirm the API & its updated category

File: tools/apidoc/gen_toc.py [hunk 1]
```diff
@@ -201,13 +166,15 @@
...
     'Condition': 'AutoScale',
     'Api': 'API Discovery',
+    'ApiLimit': 'Configuration',
     'Region': 'Region',
     'Detail': 'Resource metadata',
     'addIpToNic': 'Nic',
     'removeIpFromNic': 'Nic',
     'updateVmNicIp': 'Nic',
     'listNics':'Nic',
 	'AffinityGroup': 'Affinity Group',
```
Comment (DaanHoogland) [inline tools/apidoc/gen_toc.py:176]: isn't a crooked indentation a syntax error in python :exploding_head: ???

**Review by DaanHoogland — APPROVED** [review]
if the indentation issue is not a problem for pythonc : clgtm

File: tools/apidoc/gen_toc.py (hunk 1 above)
Comment (DaanHoogland) [reply tools/apidoc/gen_toc.py:176]:
```suggestion
    'AffinityGroup': 'Affinity Group',
```

**Comment by vishesh92** [comment]
@blueorangutan package

**Comment by sureshanaparti** [comment]
@JoaoJandre I think, it's better to have this in 4.20, as the APIs would have the updated/correct categories in the docs.

**Comment by rohityadavcloud** [comment]
Agree @sureshanaparti could be useful as usually the major release api docs is what we upload to project website. Let me help QA this.

**Comment by rohityadavcloud** [comment]
Manually built and tested - OK. API docs LGTM. Screenshot below, passing to @JoaoJandre to consider and merge:

image

**Pull request merged**